src/Pkg.jl

```diff
@@ -612,6 +612,16 @@ Upgrades the format of the manifest file from v1.0 to v2.0 without re-resolving.
 """
 const upgrade_manifest = API.upgrade_manifest
 
+"""
+    is_manifest_current(ctx::Context = Context())
+
+Returns whether the active manifest was resolved from the active project state.
+For instance, if the project had compat entries changed, but the manifest wasn't re-resolved, this would return false.
+
+If the manifest doesn't have the project hash recorded, `nothing` is returned.
+"""
+const is_manifest_current = API.is_manifest_current
```

Comment:
Perhaps this can simply be indicated by Pkg.status instead of a separate API function?

Reply:
Adding a warning to `Pkg.status` for visual review is a good idea. But one purpose of exposing this is to return a testable Bool. Perhaps we could make `Pkg.status(test_manifest_current = true)` return a Bool, but that seems like cramming functionality in.

Reply:
Who will use this function and when? What do you do with the result?

Reply:
During CI you could choose to check `@test Pkg.is_manifest_current()`.

Alternatively we could go with @DilumAluthge's suggestion of turning the `Pkg.instantiate()` warning into an error if `ENV["CI"]` is detected?

Reply:
An alternative name is `Pkg.is_environment_synced()`

Reply:
@fredrikekre what do you think?

Reply:
Can we unblock this?

Reply:
It's just an internal function for now so let's just go with this for now and we can decide later if it is put in the manual.